index.html

```diff
@@ -134,247 +134,142 @@
-       <code><a href="http://www.w3.org/TR/navigation-timing-2/#performancetiming">PerformanceTiming</a></code> interface [[!navigation-timing-2]].
+partial interface Performance {
+    void mark(DOMString markName);
+    void clearMarks(optional DOMString markName);
+
+    void measure(DOMString measureName, optional DOMString startMark, optional DOMString endMark);
+    void clearMeasures(optional DOMString measureName);
+};
+</pre>
+  <p>
+    All <a href='http://www.w3.org/TR/html51/webappapis.html#global-object'>global object</a> [[!HTML51]] have a <dfn>mark list</dfn>. Unless the <a for='Performance'>mark</a> is invoked, its <a>mark list</a> is empty. Each member of the list is a <a>PerformanceMark</a> object.
+  </p>
+
+  <p>
+    All <a href='http://www.w3.org/TR/html51/webappapis.html#global-object'>global object</a> have a <dfn>measure list</dfn>. Unless the <a for='Performance'>measure</a> is invoked, its <a>measure list</a> is empty. Each member of the list is a <a>PerformanceMeasure</a> object.
+  </p>
+
+  <p>The <dfn for="Performance" data-lt='mark'>mark(markName)</dfn> method stores a timestamp with the associated name (a "mark"). It MUST run these steps:</p>
+  <ol>
+    <li>If <var>markName</var> uses the same name as an attribute in the
+        <code><a href="http://www.w3.org/TR/navigation-timing-2/#idl-def-PerformanceTiming">PerformanceTiming</a></code> interface [[!NAVIGATION-TIMING-2]], <a href='https://heycam.github.io/webidl/#dfn-throw'>throw</a> a <a href='https://heycam.github.io/webidl/#syntaxerror'><code>SyntaxError</code></a>.</li>
+    <li>Create a new <a>PerformanceMark</a> object.</li>
+    <li>Set <code>name</code> to <var>markName</var>.</var>
+    <li>Set <code>entryType</code> to <code>DOMString "mark"</code>.
+    <li>Set <code>startTime</code> to the time in milliseconds from the <a href='http://w3c.github.io/hr-time/#time-origin'>time origin</a> to the occurrence of the call of the <a for='Performance'>mark</a> method.</li>
+    <li>Set <code>duration</code> to <code>0</code>.</li>
+    <li><a href="https://w3c.github.io/performance-timeline/#dfn-queue-a-performanceentry">Queue</a> the <a>PerformanceMark</a> object.</li>
+    <li id='stored_mark'>Add the <a>PerformanceMark</a> object to <a>mark list</a>.</li>
+    <li>Return <strong>undefined</strong>.</li>
+  </ol>
+
+  <p>The <dfn for="Performance" data-lt='clearMarks'>clearMarks(markName)</dfn> removes stored timestamp with the associated name. It MUST run these steps:</p>
+  <ol>
+    <li>Let <var>set</var> be a new empty <a href='http://heycam.github.io/webidl/#idl-sequence'>sequence</a>.</li>
+    <li>If <var>markName</var> is omitted, add to <var>set</var> all <a>PerformanceMark</a> objects in <a>mark list</a>. Otherwise, add to <var>set</var> all <a>PerformanceMark</a> objects with a <code>name</code> attribute value equals to the value <var>markName</var> in <a>mark list</a>.</li>

```

Comment:
The `set` business feels redundant, no? 

If markName is omitted, clear mark list. Otherwise, remove all objects listed in `mark list` whose name matches `markName`?